src/deephaven/plugin/matplotlib/__init__.py

```diff
@@ -16,6 +19,141 @@ def init_theme():
+            ret = func(update, *args, **kwargs)
+            if ret == 0:
+                self.remove_callback(func)
+
+        if len(self.callbacks) == 0:
+            self.stop()
+
+    def start(self):
+        if self._listener is None:
+            self._listener = listen(self._table, self._on_table_update)
+
+    def stop(self):
+        if self._listener is not None:
+            self._listener.stop()
+            self._listener = None
+
+class TableAnimation(Animation):
+    """
+    Makes an animation by calling a function *func* whenever the Deephaven Table *table* is updated.
+
+    .. note::
+
+        You must store the created Animation in a variable that lives as long
+        as the animation should run. Otherwise, the Animation object will be
+        garbage-collected and the animation stops.
+
+    fig : `~matplotlib.figure.Figure`
+        The figure object used to get needed events, such as draw or resize.
+
+    table : `~deephaven.Table`
+        The table object used to listen for updates
+
+    func : callable
+        The function to call at each update. The first argument is a dictionary
+        with the new data arrays, the second argument is the update that
+        triggered the function. On the first call `update` will be `None`.
+        Any additional positional arguments can be supplied via the *fargs*
+        parameter.
+
+        The required signature is::
+
+            def func(data, update, *fargs) -> None
+
+    columns : tuple or None, optional
+        Names of columns to send along in the update, or None to send all column data
+
+    fargs : tuple or None, optional
+        Additional arguments to pass to each call to *func*.
+
+    """
+    def __init__(self, fig, table, func, columns=None, fargs=None, **kwargs):
+        if fargs:
+            self._args = fargs
+        else:
+            self._args = ()
+        self._func = func
+        self._table = table
+        if not columns:
+            self._columns = [column.name for column in table.columns]
+        else:
+            self._columns = columns
+        self._last_update = None
+        event_source = TableEventSource(table)
+        super().__init__(fig, event_source, **kwargs)
+
+        # Start the animation right away
+        self._start()
+
+    def new_frame_seq(self):
+        """
+        new_frame_seq
+        """
+        # Use the generating function to generate a new frame sequence
+        return itertools.count()
+
+    def _step(self, update, *args):
+        """Handler for getting events."""
+        # Extends the _step() method for the Animation class. Used
+        # to get the update information
+        self._last_update = update
+        return super()._step(*args)
+
+    def _draw_frame(self, framedata):
+        data = {}
+        for column in self._columns:
+            data[column] = self._table.j_table.getColumn(column).getDirect()
+        self._func(data, self._last_update, *self._args)
```

Comment:
The new Table listener interface provides the updates in a dict with key being the column name and the value being a numpy arrray. Not sure if it makes sense to use it for efficiency purpose

Reply:
Yea, it depends on how the data is used in matplotlib. Different charts use it different. As such I decided to provide both (direct data and the update).